src/App.jsx

```diff
@@ -58,7 +66,41 @@ const App = () => {
         "Failed to determine target origin or iframe is not available"
       );
     }
-  }
+  };
+
+  const handleIframeLoad = (iframe) => {
+    const iframeWindow = iframe.contentWindow;
+
+    const messageListener = (event) => {
+      if (event.origin !== new URL(iframe.src).origin) return;
+      if (event.data === "appLoaded" && theme) {
+        window.removeEventListener("message", messageListener);
+        sendMessageToIframe(iframe, { customThemeColors: theme });
+      }
+    };
+
+    iframeWindow.addEventListener("load", () => {
+      const leftNav = iframe.contentDocument.getElementById("left-nav");
+      if (leftNav) {
+        iframeWindow.postMessage("appLoaded", new URL(iframe.src).origin);
+      }
+    });
+
+    window.addEventListener("message", messageListener);
+  };
+
+  useEffect(() => {
+    theme && sendColorSettingsToIframes();
+  }, [activeApp.link, showConnect, theme]);
+
+  const sendColorSettingsToIframes = () => {
+    const message = { customThemeColors: theme };
```

Comment:
the theme should not be sent to the child UI's unless it is a custom theme loaded from theme.json. create a separate useState for the custom theme.